cmake/NSIS.template.in

```diff
@@ -937,9 +937,123 @@ SectionEnd
+  "Cura 15.04.4 is already installed. $\n$\nDo you want to uninstall the old version before installing the new one?" \
+  IDYES uninst_15044 IDNO lookFor15045
+
+uninst_15044:
+  ClearErrors
+  StrLen $2 "\uninstall.exe"
+  StrCpy $3 $0 -$2 # remove "\uninstall.exe" from UninstallString to get path
+  ExecWait '$0 _?=$3' ;Do not copy the uninstaller to a temp file
+
+  IfErrors uninst_failed lookFor15045
+
+lookFor15045:  
+  ; Cura 15.04.5
+  ReadRegStr $0 HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Cura_15.04.5" "UninstallString"
+  ${CharStrip} '"' $0 $0
+  StrCmp $0 "" lookFor15046
+
+  MessageBox MB_YESNOCANCEL|MB_ICONEXCLAMATION \
+  "Cura 15.04.5 is already installed. $\n$\nDo you want to uninstall the old version before installing the new one?" \
+  IDYES uninst_15045 IDNO lookFor15046
+
+uninst_15045:
+  ClearErrors
+  StrLen $2 "\uninstall.exe"
+  StrCpy $3 $0 -$2 # remove "\uninstall.exe" from UninstallString to get path
+  ExecWait '$0 _?=$3' ;Do not copy the uninstaller to a temp file
+
+  IfErrors uninst_failed lookFor15046
+
+lookFor15046:  
+  ; Cura 15.04.6
+  ReadRegStr $0 HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Cura_15.04.6" "UninstallString"
+  ${CharStrip} '"' $0 $0
+  StrCmp $0 "" lookFor21
+
+  MessageBox MB_YESNOCANCEL|MB_ICONEXCLAMATION \
+  "Cura 15.04.6 is already installed. $\n$\nDo you want to uninstall the old version before installing the new one?" \
+  IDYES uninst_15046 IDNO lookFor21
 
+uninst_15046:
+  ClearErrors
+  StrLen $2 "\uninstall.exe"
+  StrCpy $3 $0 -$2 # remove "\uninstall.exe" from UninstallString to get path
+  ExecWait '$0 _?=$3' ;Do not copy the uninstaller to a temp file
+
+  IfErrors uninst_failed lookFor21
+
+lookFor21:  
+  ; Cura 2.1
+  ReadRegStr $0 HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Cura 2.1" "UninstallString"
+  StrCmp $0 "" lookFor23
+
+  MessageBox MB_YESNOCANCEL|MB_ICONEXCLAMATION \
+  "Cura 2.1 is already installed. $\n$\nDo you want to uninstall the old version before installing the new one?" \
+  IDYES uninst_21 IDNO lookFor23
+
+uninst_21:
+  ClearErrors
+  StrLen $2 "\Uninstall.exe"
+  StrCpy $3 $0 -$2 # remove "\Uninstall.exe" from UninstallString to get path
+  ExecWait '$0 _?=$3' ;Do not copy the uninstaller to a temp file
+
+  IfErrors uninst_failed lookFor23
+
+lookFor23:  
+  ; Cura 2.3
+  ReadRegStr $0 HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Cura 2.3" "UninstallString"
+  StrCmp $0 "" lookForCurrentVersion
+
+  MessageBox MB_YESNOCANCEL|MB_ICONEXCLAMATION \
+  "Cura 2.3 is already installed. $\n$\nDo you want to uninstall the old version before installing the new one?" \
+  IDYES uninst_23 IDNO lookForCurrentVersion
+
+uninst_23:
+  ClearErrors
+  StrLen $2 "\Uninstall.exe"
+  StrCpy $3 $0 -$2 # remove "\Uninstall.exe" from UninstallString to get path
+  ExecWait '$0 _?=$3' ;Do not copy the uninstaller to a temp file
+
+  IfErrors uninst_failed lookForCurrentVersion
+
+lookForCurrentVersion:
```

Comment:
Several releases are still missing.

I would also hope that we won't need to modify this code every time we make a release though. Currently it seems to work automatically. Why was this necessary for updating the Arduino drivers?

Reply:
I don't know. It is the same question like: Why shall I install updates and upgrades my computer? :roll_eyes: